extensions/narayana-jta/deployment/src/main/java/io/quarkus/narayana/jta/deployment/NarayanaJtaProcessor.java

```diff
@@ -51,9 +51,16 @@
     @Record(RUNTIME_INIT)
     public void build(NarayanaJtaRecorder recorder, BuildProducer<FeatureBuildItem> feature) {
         feature.produce(new FeatureBuildItem(FeatureBuildItem.NARAYANA_JTA));
+
         additionalBeans.produce(new AdditionalBeanBuildItem(NarayanaJtaProducers.class));
+
         runtimeInit.produce(new RuntimeInitializedClassBuildItem(
                 "com.arjuna.ats.internal.jta.resources.arjunacore.CommitMarkableResourceRecord"));
+        runtimeInit.produce(new RuntimeInitializedClassBuildItem("com.arjuna.ats.arjuna.coordinator.TxControl"));
+        runtimeInit.produce(new RuntimeInitializedClassBuildItem("com.arjuna.ats.arjuna.recovery.ActionStatusService"));
```

Comment:
the `com.arjuna.ats.arjuna.recovery.ActionStatusService` doesn't seem to have static initializers. Did you mean to force re-initialization to null of the one field instead? I don't think that's accomplished with this specific build item.

Reply:
I added it at some point because otherwise native-image complained for some obscure reasons. I made several iterations though so maybe it's not needed in the end.

Reply:
What happens if you comment it? Does it still work?

Reply:
I'll try.

Reply:
I got no complaints.. I guess it's no longer necessary.  But I'll try to see if that static field needs resetting.

How confident are you that you tracked down all references to `Uid` s? Do we have a way to ensure there's none in the captured heap? I kinda remember doing such checks but don't remember if we had a way to do it in an integration test.